jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/ExtensionConfig.java

```diff
@@ -343,45 +348,26 @@ else if (paramNameIdx > 0)
                 paramName = buffer.substring(paramNameIdx);
             }
 
-            Map<String, String> paramMap = getParamMap(extName);
-            paramMap.put(paramName, paramValue);
-
+            params.put(paramName, paramValue);
             super.parsedParam(buffer, valueLength, paramNameIdx, paramValueIdx);
         }
 
         @Override
         protected void parsedValue(StringBuffer buffer)
         {
             String extName = buffer.toString();
-            getParamMap(extName);
+            if (name != null)
+                throw new IllegalArgumentException("parameterizedName contains multiple ExtensionConfigs: " + parameterizedName);
+            name = extName;
             super.parsedValue(buffer);
         }
 
-        private Map<String, String> getParamMap(String extName)
+        public ParamParser parse()
```

Comment:
Why returning `ParamParser` here? Should not this just be `void`?
The return value is used only once, so perhaps you want to do something different: 
```java
public static ParamParser parse(parameterizedName)
```
The key point is that you are conflating 2 things here: the parsing and the result of the parsing, using the same class for both, which is kind of confusing - hence the `static` method.